pages/index.tsx

```diff
@@ -61,51 +64,64 @@ const Home: React.FC<HomeProps> = ({
   const [folders, setFolders] = useState<Folder[]>([]);
 
   const [conversations, setConversations] = useState<Conversation[]>([]);
-  const [selectedConversation, setSelectedConversation] =
-    useState<Conversation>();
-  const [currentMessage, setCurrentMessage] = useState<Message>();
+  
+  // const [selectedConversation, setSelectedConversation] =
+  //   useState<Conversation>();
+  // const [currentMessage, setCurrentMessage] = useState<Message>();
 
   const [showSidebar, setShowSidebar] = useState<boolean>(true);
 
   const [prompts, setPrompts] = useState<Prompt[]>([]);
   const [showPromptbar, setShowPromptbar] = useState<boolean>(true);
 
+  const {
+    currentMessageList,
+    selectedConversation,
+    setSelectedConversation,
+    actions,
+  } = useContext(ConversationContext);
   // REFS ----------------------------------------------
 
   const stopConversationRef = useRef<boolean>(false);
 
   // FETCH RESPONSE ----------------------------------------------
 
-  const handleSend = async (message: Message, deleteCount = 0) => {
+  const handleSend = async (
+    chatNode: ChatNode,
+    sendAction: SendAction,
+    messageIndex?: number,
+  ) => {
     if (selectedConversation) {
       let updatedConversation: Conversation;
-
-      if (deleteCount) {
-        const updatedMessages = [...selectedConversation.messages];
-        for (let i = 0; i < deleteCount; i++) {
-          updatedMessages.pop();
-        }
-
-        updatedConversation = {
-          ...selectedConversation,
-          messages: [...updatedMessages, message],
-        };
-      } else {
-        updatedConversation = {
-          ...selectedConversation,
-          messages: [...selectedConversation.messages, message],
-        };
+      let sendMessages: Message[];
```

Comment:
The type of `Message` is causing the error when sending the request.
```
[OpenAIError: Additional properties are not allowed ('create_time', 'id' were unexpected) - 'messages.1'] {
  name: 'OpenAIError',
  type: 'invalid_request_error',
  param: null,
  code: null
}
```

Reply:
I add another type to fixed it.
https://github.com/littlemoonstones/chatbot-ui/blob/8487bfa2e6c58d7388e62dfe07ecb704d746c8d0/types/chat.ts#L4-L14

Reply:
Thanks @littlemoonstones! I'll merge that in today.

Reply:
Fixed